docs/source/howto/workflows.rst

```diff
@@ -18,7 +18,159 @@ Writing workflows
+
+.. code-block:: python
+
+    from aiida.engine import run
+    from aiida.plugins import WorkflowFactory, DataFactory
+    Int = DataFactory('int')
+    MultiplyAndAddWorkChain = WorkflowFactory('arithmetic.multiply_add')
+
+    add_code = load_code(label='add')
+
+    results = run(MultiplyAndAddWorkChain, x=Int(2), y=Int(3), z=Int(5), code=add_code)
+
+Alternatively, you can first construct a dictionary of the inputs, and pass it to the ``run`` function by taking advantage of `Python's automatic keyword expansion <https://docs.python.org/3/tutorial/controlflow.html#unpacking-argument-lists>`_:
+
+.. code-block:: python
+
+    inputs = {'x': Int(1), 'y': Int(2), 'z': Int(3), 'code': add_code}
+    results = run(MultiplyAndAddWorkChain, **inputs)
+
+This is particularly useful in case you have a workflow with a lot of inputs.
+In both cases, running the ``MultiplyAddWorkChain`` workflow returns the **results** of the workflow, i.e. a dictionary of the nodes that are produced as outputs, where the keys of the dictionary correspond to the labels of each respective output.
+
+.. note::
+
+    Similar to other processes, there are multiple functions for launching a work chain.
+    See the section on :ref:`launching processes for more details<topics:processes:usage:launching>`.
+
+Since *running* a workflow will block the interpreter, you will have to wait until the workflow is finished before you get back control.
+Moreover, you won't be able to turn your computer or even your terminal off until the workflow has fully terminated, and it is difficult to run multiple workflows in parallel.
+So, it is advisable to *submit* more complex or longer work chains to the daemon:
+
+.. code-block:: python
+
+    from aiida.engine import submit
+    Int = DataFactory('int')
+    MultiplyAndAddWorkChain = WorkflowFactory('arithmetic.multiplyadd')
+
+    add_code = load_code(label='add')
+    inputs = {'x': Int(1), 'y': Int(2), 'z': Int(3), 'code': add_code}
+
+    workchain_node = submit(MultiplyAndAddWorkChain, **inputs)
+
+Note that when using ``submit`` the work chain is not run in the local interpreter but is sent off to the daemon and you get back control instantly.
+This allows you to submit multiple work chains at the same time and the daemon will start working on them in parallel.
+Once the ``submit`` call returns, you will not get the result as with ``run``, but you will get the **node** that represents the work chain.
+This is because at that point in time, the results do not exist yet, because the work chain has not finished.
+In addition to making it easy to execute multiple work chains in parallel, submitting work chains also allows you to shut down your computer.
+The daemon will always save the progress of the work chains that it is managing such that if it is stopped, it can restart where it left off.
```

Comment:
```suggestion
Submitting a work chain instead of directly running it not only makes it easier to execute multiple work chains in parallel, but also ensures that the progress of a workchain is not lost when you restart your computer.
```